tests/queries/0_stateless/00718_format_datetime.sql

```diff
@@ -40,6 +40,10 @@ SELECT formatDateTime(toDateTime('2018-01-02 22:33:44'), '%%');
 SELECT formatDateTime(toDateTime('2018-01-02 22:33:44'), 'no formatting pattern');
 
 SELECT formatDateTime(toDate('2018-01-01'), '%F %T');
+
+SELECT formatDateTime(toDate32('2018-01-01'), '%F %T');
```

Comment:
If I understand correctly, this PR also introduces Date32 support for `FROM_UNIXTIME`/
`fromUnixTimestamp`. Therefore, would you like to add a short test for that, e.g. to 01411_from_unixtime.sql?

Reply:
In this test (00718), it would actually be nicer to test for each given format specifier (l. 11-40) that it works for all argument types, e.g. in l. 11, we could test

```
SELECT formatDateTime(toDateTime('2018-01-02 22:33:44'), '%C'), formatDateTime(toDate32('2018-01-02 22:33:44'), '%C');
```

Reply:
> If I understand correctly, this PR also introduces Date32 support for `FROM_UNIXTIME`/
> `fromUnixTimestamp`.
I have made a quick code review of functions `FROM_UNIXTIME`/ `fromUnixTimestamp` these functions does not use `formatDateTime` for it's implementation.

Reply:
I thought it did because of

```
using FunctionFormatDateTime = FunctionFormatDateTimeImpl<NameFormatDateTime, false>;
using FunctionFromUnixTimestamp = FunctionFormatDateTimeImpl<NameFromUnixTime, true>;

REGISTER_FUNCTION(FormatDateTime)                               
{                                                               
    factory.registerFunction<FunctionFormatDateTime>();         
    factory.registerFunction<FunctionFromUnixTimestamp>();      
    factory.registerAlias("FROM_UNIXTIME", "fromUnixTimestamp");
}                                                               
```

in src/Functions/formatDateTime.cpp and because you added a check for `Date32` in l. 327 (in the new version) which affects `support_integer == true` which is for `fromUnixTimestamp`/`FROM_UNIXTIME`.

Reply:
Very interesting. I have not noticed `FunctionFormatDateTimeImpl` usage from my first look. I will make some analysis on it.

Reply:
Robert, I verified it by both more careful code analysis and by running CH from master and from my branch with queries like:

`SELECT FROM_UNIXTIME(toDate32('2022-10-20'), '%F %T') AS DateTime;`

So I confirm: it this PR we also add support for Date32 arguments for `FROM_UNIXTIME` function too. I will change it in PR's title and changelog.

Reply:
> In this test (00718), it would actually be nicer to test for each given format specifier (l. 11-40) that it works for all argument types,

OK. One question: should we also add tests for patterns like %H (hour) and other kinds of patterns for hours, minutes and seconds? Because it does not make sense for Date32.

Reply:
OK. I decided to add it for hours, minutes, seconds too.

Reply:
@rschu1ze I added Date32 to 01411_from_unixtime and improved 00718_format_datetime. Please take a look.
BTW, in your example:

`SELECT formatDateTime(toDateTime('2018-01-02 22:33:44'), '%C'), formatDateTime(toDate32('2018-01-02 22:33:44'), '%C');`

`toDate32` does not accept arguments like 2018-01-02 22:33:44. Only dates are allowed: 2018-01-02
That's why in my tests I don't run formatDateTime(toDate32()) twice to the same date.

Reply:
Thanks, I am fine with tests for hours, minutes and seconds.

Sorry for if the example was misleading, I did not try it out...